pkg/controller/deliverable/reconciler.go

```diff
@@ -90,7 +93,24 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	}
 	r.conditionManager.AddPositive(DeliveryReadyCondition())
 
-	stampedObjects, err := r.Realizer.Realize(ctx, realizer.NewResourceRealizer(deliverable, r.Repo, delivery.Spec.Params), delivery)
+	serviceAccountName := "default"
+	if deliverable.Spec.ServiceAccountName != "" {
+		serviceAccountName = deliverable.Spec.ServiceAccountName
+	}
+
+	secret, err := r.Repo.GetServiceAccountSecret(ctx, serviceAccountName, deliverable.Namespace)
+	if err != nil {
+		r.conditionManager.AddPositive(ServiceAccountSecretNotFoundCondition(err))
+		return r.completeReconciliation(ctx, deliverable, fmt.Errorf("get secret for service account '%s': %w", deliverable.Spec.ServiceAccountName, err))
+	}
+
+	resourceRealizer, err := r.ResourceRealizerBuilder(secret, deliverable, r.Repo, delivery.Spec.Params)
+	if err != nil {
+		r.conditionManager.AddPositive(ResourceRealizerBuilderErrorCondition(err))
+		return r.completeReconciliation(ctx, deliverable, controller.NewUnhandledError(fmt.Errorf("build resource realizer: %w", err)))
```

Comment:
with new error and logging this should be 
`failed to build resource realizer: %w`